Mathlib/GroupTheory/FreeProd/Basic.lean

```diff
@@ -0,0 +1,344 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Yury Kudryashov
+-/
+import Mathlib.GroupTheory.Subgroup.Basic
+import Mathlib.GroupTheory.Congruence
+import Mathlib.GroupTheory.Submonoid.Membership
+
+/-!
+# Free product of two monoids or groups
```

Comment:
This should include a comparison to `FreeProduct`

Reply:
Which should probably be renamed to `FreeProdI` or `FreeProdᵢ` while porting.

Reply:
`pi_free_product` would also be a good choice, since that matches `pi_tensor_product` vs `tensor_product`.

Reply:
Note that the file is a way from being ported in mathlib3; I would prefer we rename it there if we plan to rename it.

Reply:
Renamed in leanprover-community/mathlib#18437